extension/src/main/java/com/mobven/extension/IntentExt.kt

```diff
@@ -0,0 +1,126 @@
+package com.mobven.extension
+
+import android.app.Activity
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+
+fun Intent.clearTask() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
+}
+
+fun Intent.clearTop() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+}
+
+fun Intent.clearWhenTaskReset() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT)
+}
+
+fun Intent.excludeFromRecents() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS)
+}
+
+fun Intent.multipleTask() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
+}
+
+fun Intent.newTask() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+}
+
+fun Intent.noAnimation() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
+}
+
+fun Intent.noHistory() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
+}
+
+fun Intent.restart() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+}
+
+fun Intent.singleTop() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+}
+
+fun Intent.reorderToFront() = apply {
+    addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
+}
+
+fun Intent.canBeHandled(context: Context) = this.resolveActivity(context.packageManager) != null
+
+fun Context.intentCanBeHandled(intent: Intent) = intent.resolveActivity(packageManager) != null
+
+
+fun Context.getActivityPendingIntent(
+    requestCode: Int = 0,
+    intent: Intent,
+    flags: Int = PendingIntent.FLAG_ONE_SHOT
+): PendingIntent =
+    PendingIntent.getActivity(this, requestCode, intent, flags)
+
+fun Context.getBroadcastPendingIntent(
+    requestCode: Int = 0,
+    intent: Intent,
+    flags: Int = PendingIntent.FLAG_ONE_SHOT
+): PendingIntent =
+    PendingIntent.getBroadcast(this, requestCode, intent, flags)
+
+fun Context.getProductionApplicationId(): String {
+    val applicationId = packageName
+    return when {
+        applicationId.contains(".stage") -> applicationId.dropLast(6)
```

Comment:
I think, these variants need to be generic. It could be getProductionApplicationId(variant: String, dropLast: Int)